packages/headless/src/features/commerce/common/actions.ts

```diff
@@ -71,7 +76,7 @@ export const buildCommerceAPIRequest = async (
 ): Promise<CommerceAPIRequest> => {
   const facets = getFacets(state);
   return {
-    ...buildBaseCommerceAPIRequest(state),
+    ...(await buildBaseCommerceAPIRequest(state)),
```

Comment:
Spotted this little bug where we build requests. fyi @BenKandelaars I think this might be related to your issue?